docs/runtime/types/module-struct.md

```diff
@@ -0,0 +1,241 @@
+impl<T: Trait> Module<T> {
+    pub fn do_something(origin: T::Origin, something: u32) -> Result {
+        let who = ensure_signed(origin)?;
+        <Something<T>>::put(something);
+        Self::deposit_event(RawEvent::SomethingStored(something, who));
+        Ok(())
+    }
+}
+```
+
+`Module` struct implements the `Callable` trait in `srml-support` by specifying the associate type to be the module's `Call` enum. Above dispatchable functions will be exposed through this enum, go to its own page for more details about [`Call` enum]((runtime/types/call-enum.md)).
+
+```rust
+impl<T: Trait> ::srml_support::dispatch::Callable for Module<T> {
+    type Call = Call<T>;
+}
+```
+
+There is also a `dispatch` function that redirect the request to the module's `Call` enum:
+
+```rust
+impl<T: Trait> Module<T> {
+    #[doc(hidden)]
+    pub fn dispatch<D: ::srml_support::dispatch::Dispatchable<Trait=T>>(
+        d: D,
+        origin: D::Origin,
+    ) -> ::srml_support::dispatch::Result {
+        d.dispatch(origin)
+    }
+}
+```
+
+### Implementations for Storage
+
+The input codes for the `decl_storage!` procedure macro in template module are:
+
+```rust
+/// This module's storage items.
+decl_storage! {
+    trait Store for Module<T: Trait> as TemplateModule {
+        // Just a dummy storage item. 
+        // Here we are declaring a StorageValue, `Something` as a Option<u32>
+        // `get(something)` is the default getter which returns either the stored `u32` or `None` if nothing stored
+        Something get(something): Option<u32>;
+    }
+}
+```
+
+After expanded, we'll see a `Store` implementation for `Module` struct. The `Store` trait here is an abstract layer that places all the associate types for module's available storage items. `Module` struct implements this trait by specifying the concrete types like `Something<T>` struct.
+
+```rust
+trait Store {
+    type Something;
+}
+
+struct Something<T: Trait>(
+    self::sr_api_hidden_includes_decl_storage::hidden_include::storage::generator::PhantomData<
+        (T),
+    >,
+);
+
+impl<T: Trait> Store for Module<T> {
+    type Something = Something<T>;
+}
+```
+
+The `Module` struct also implements the default getter functions for storage items and the query for storage metadata. Go to [Metadata of Your Module](#metadata-of-your-module) section to give a glimpse at details. 
+
+### Deposit Event in Module
+
+Remember the `deposit_event` function defined in the `decl_module!` macro? It will call the `deposit_event` function in `srml-system` runtime module to deposit an event into block's event record.
+
+```rust
+impl<T: Trait> Module<T> {
+    fn deposit_event(event: Event<T>) {
+        <system::Module<T>>::deposit_event(<T as Trait>::from(event).into());
+    }
+}
+```
+
+Refer to [Event Enum](runtime/types/event-enum.md) for more knowledge around events in Substrate.
+
+### Utilities for Block Execution
+
+There is three privileged functions that can be define in runtime modules:
```

Comment:
Don't use the word "privileged" here. In Substrate, "privileged functions" are those which require ROOT origin.

I would say there are 3 "reserved" functions or "functions with reserved names"